sql/updates/latest-dev.sql

```diff
@@ -172,6 +172,48 @@ BEGIN
     WHERE mat_hypertable_id OPERATOR(pg_catalog.=) _cagg_data.mat_hypertable_id;
 END;
 $BODY$;
+CREATE TABLE _timescaledb_internal.job_errors (
+  job_id integer not null, 
+  pid integer,
+  start_time timestamptz,
+  finish_time timestamptz,
+  error_data jsonb
+);
+
+SELECT pg_catalog.pg_extension_config_dump('_timescaledb_internal.job_errors', '');
+
+CREATE VIEW timescaledb_information.job_errors AS 
+SELECT
+    job_id,
+    error_data ->> 'proc_schema' as proc_schema,
+    error_data ->> 'proc_name' as proc_name,
+    pid,
+    start_time,
+    finish_time,
+    error_data ->> 'sqlerrcode' AS sqlerrcode,
+    CASE WHEN error_data ->>'message' IS NOT NULL THEN 
+      CASE WHEN error_data ->>'detail' IS NOT NULL THEN 
+        CASE WHEN error_data ->>'hint' IS NOT NULL THEN concat(error_data ->>'message', '. ', error_data ->>'detail', '. ', error_data->>'hint')
+        ELSE concat(error_data ->>'message', ' ', error_data ->>'detail')
+        END
+      ELSE
+        CASE WHEN error_data ->>'hint' IS NOT NULL THEN concat(error_data ->>'message', '. ', error_data->>'hint')
+        ELSE error_data ->>'message'
+        END
+      END
+    ELSE
+      'job crash detected, see server logs'
+    END
+    AS err_message
+FROM
+    _timescaledb_internal.job_errors;
+
+ALTER TABLE _timescaledb_internal.bgw_job_stat ADD COLUMN flags integer;
+UPDATE _timescaledb_internal.bgw_job_stat SET flags = 0;
```

Comment:
```suggestion
ALTER TABLE _timescaledb_internal.bgw_job_stat ADD COLUMN flags integer NOT NULL DEFAULT 0;
```

This should add the `NOT NULL` constraint during upgrades. It should also make sure the column is initialized without extra overhead at `ALTER TABLE` statement time.

Reply:
Hmm, apparently, this doesn't work as per the comment in `test/sql/updates/catalog_missing_columns.sql` so I think I'll have to do it as before.